src/model.py

```diff
@@ -113,6 +115,11 @@ def initialize(self, args):
         # Counter to keep track of ongoing request counts
         self.ongoing_request_count = 0
 
+        # Starting the response thread
+        self._response_queue = queue.Queue()
+        self._response_thread = threading.Thread(target=self.response_loop)
```

Comment:
Was the bottleneck in the response callback? So, this is a solution to unlock async engine execution to proceed while the response callback is being invoked in a different thread?

Does it make sense to push this optimization into the python backend response sender implementation?  A separate response thread reading and returning response callback? 
On the model.py side it would simply become an enqueue to the response queue which is maintained in python backend?

Reply:
> Was the bottleneck in the response callback? So, this is a solution to unlock async engine execution to proceed while the response callback is being invoked in a different thread?

Correct.

> Does it make sense to push this optimization into the python backend response sender implementation? A separate response thread reading and returning response callback?
On the model.py side it would simply become an enqueue to the response queue which is maintained in python backend?

We do have longer term plans to support something similar using `async_send`. The immediate issue with what you're proposing is that we need to tell the user whether there has been an error with sending the response or not. If we put the user's request into an internal queue and declare success, there is no way to communicate the error message.

Reply:
Thanks for the explanation!